File: tmp-mei/fgen4255027576711118260/base/Error_0.java
package base;
public interface Error_0{
Error_0 $self = new Error_0Impl();
Object $exclamation$imm(base.Info_0 info_m$);

Object msg$imm(rt.Str msg_m$);
static Object $exclamation$imm$fun(base.Info_0 info_m$, base.Error_0 $this) {
  return (switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (Object) null;
}})
;
}

static Object msg$imm$fun(rt.Str msg_m$, base.Error_0 $this) {
  return (switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (Object) null;
}})
;
}
}

File: tmp-mei/fgenjs5030319945734616766/base/Error_0.js
import * as rt from "../rt/main.js";
export class base$$Error_0 {
  static $self = new base$$Error_0();

  $exclamation$imm(info_m$) {
  return base$$Error_0.$exclamation$imm$fun(info_m$, this);
}


msg$imm(msg_m$) {
  return base$$Error_0.msg$imm$fun(msg_m$, this);
}

  static $exclamation$imm$fun(info_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static msg$imm$fun(msg_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

}
